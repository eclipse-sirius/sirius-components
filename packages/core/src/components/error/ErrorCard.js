/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/

import React from 'react';
import PropTypes from 'prop-types';

import { classNames } from '../../common/classnames';

import { Card } from '../cards/Card';
import { Spacing } from '../spacing/Spacing';
import { M, L } from '../spacing/SpacingConstants';

import './ErrorCard.css';

const propTypes = {
  code: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired
};

const ERRORCARD__CLASS_NAMES = 'errorcard';
const ERRORCARD_CONTAINER__CLASS_NAMES = 'errorcard-container';
const ERRORCARD_TITLE__CLASS_NAMES = 'title-l';

/**
 * The ErrorCard component is used to display errors.
 */
export const ErrorCard = ({ className, code, title, message, ...props }) => {
  const errorCardClassNames = classNames(ERRORCARD__CLASS_NAMES, className);
  return (
    <Card {...props} className={errorCardClassNames}>
      <Spacing top={L} right={L} bottom={L} left={L}>
        <div className={ERRORCARD_CONTAINER__CLASS_NAMES}>
          <Spacing top={M} right={M} bottom={M} left={M}>
            <h1 className={ERRORCARD_TITLE__CLASS_NAMES}>{title}</h1>
          </Spacing>
          <p>{`${message} (code: ${code})`}</p>
        </div>
      </Spacing>
    </Card>
  );
};
ErrorCard.propTypes = propTypes;
