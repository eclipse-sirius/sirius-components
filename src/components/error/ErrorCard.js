/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/

import React from 'react';
import PropTypes from 'prop-types';

import { classNames } from '../../common/classnames';

import { Card } from '../cards/Card';

import './ErrorCard.css';

const propTypes = {
  code: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired
};

const ERRORCARD__CLASS_NAMES = 'errorcard';
const ERRORCARD_TITLE__CLASS_NAMES = 'title-l';

/**
 * The ErrorCard component is used to display errors.
 */
export const ErrorCard = ({ className, code, title, message, ...props }) => {
  const errorCardClassNames = classNames(ERRORCARD__CLASS_NAMES, className);
  return (
    <Card {...props} className={errorCardClassNames}>
      <h1 className={ERRORCARD_TITLE__CLASS_NAMES}>{title}</h1>
      <p>{`${message} (code: ${code})`}</p>
    </Card>
  );
};
ErrorCard.propTypes = propTypes;
