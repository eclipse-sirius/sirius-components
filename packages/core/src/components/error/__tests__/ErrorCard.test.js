/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/

import React from 'react';
import Renderer from 'react-test-renderer';

import { ErrorCard } from '../ErrorCard';

describe('ErrorCard', () => {
  it('renders an error card', () => {
    const errorCard = Renderer.create(
      <ErrorCard
        title="An error has occurred"
        message="Please contact your administrator to find out more about this error"
        code={500}
      />
    );
    expect(errorCard.toJSON()).toMatchSnapshot();
  });
});
