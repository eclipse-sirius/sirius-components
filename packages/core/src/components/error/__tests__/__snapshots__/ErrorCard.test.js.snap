// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorCard renders an error card 1`] = `
<div
  className="card errorcard"
>
  <div
    style={
      Object {
        "paddingBottom": "32px",
        "paddingLeft": "32px",
        "paddingRight": "32px",
        "paddingTop": "32px",
      }
    }
  >
    <div
      className="errorcard-container"
    >
      <div
        style={
          Object {
            "paddingBottom": "16px",
            "paddingLeft": "16px",
            "paddingRight": "16px",
            "paddingTop": "16px",
          }
        }
      >
        <h1
          className="title-l"
        >
          An error has occurred
        </h1>
      </div>
      <p>
        Please contact your administrator to find out more about this error (code: 500)
      </p>
    </div>
  </div>
</div>
`;
